intelmq/tests/bots/parsers/shadowserver/test_download_schema.py

```diff
@@ -0,0 +1,29 @@
+# SPDX-FileCopyrightText: 2023 The Shadowserver Foundation
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Jul 27 19:44:44 2023
+
+"""
+
+import unittest
+import os
+import logging
+from intelmq import VAR_STATE_PATH
+import intelmq.bots.parsers.shadowserver._config as config
+import intelmq.lib.utils as utils
+import intelmq.lib.test as test
+
+@test.skip_internet()
+class TestShadowserverSchemaDownload(unittest.TestCase):
+
+    def test_download(self):
+        if os.path.isdir(VAR_STATE_PATH):
+            schema_file = os.path.join(VAR_STATE_PATH, 'shadowserver-schema.json')
+            config.set_logger(utils.log('test-bot', log_path=None))
+            if os.path.exists(schema_file):
+                os.unlink(schema_file)
+            self.assertEqual(True, config.update_schema())
+            self.assertEqual(True, os.path.exists(schema_file))
```

Comment:
Tests must not make any changes in the directories of IntelMQ instances. Use a [temporary directory](https://docs.python.org/3/library/tempfile.html?highlight=tempdir#tempfile.tempdir) instead

Reply:
The test now uses a temporary directory.

Reply:
Need help here.  The test includes the `@test.skip_internet()` decorator yet the automated build fails to download.  Previously I used the `os.environ.get('INTELMQ_SKIP_INTERNET')` check to handle this.

Reply:
The CI does not have `INTELMQ_SKIP_INTERNET` set because it has internet, so there's no reason to skip these tests in the CI. In other environments, internet access is not available (and thus, the result is reproducible).

Reply:
Is the CI able to create a temporary directory?

Reply:
Yes, other tests do that too

Reply:
It looks like the CI environment isn't complete:

`DEBUG    test-bot:_config.py:344 File not found: '/opt/intelmq/etc/runtime.yaml'.`

Reply:
The unittest action calls the unittests, it does not install IntelMQ. The unittests work all without an installed IntelMQ. Otherwise, they wouldn't be unittests, but integration tests or something else. What is the issue your are running into?

Reply:
The code uses the create_request_session() function from intelmq.lib.utils to support sites that use proxies.  The create_request_session function calls load_configuration which throws the exception.

Reply:
The bot may call this function. The unit test itself must not use it, except it is mocked properly (The methods to run bots in `intelmq.lib.test.TestCase` handles this). And you are missing this mocking.

What you could try is:
- create an instance of the bot with `prepare_bot` (equivalent to `run_bot`, but missing the call to `bot.proccess`)
- use the same mocking as `run_bot` to call the method you want to call:
  ```python
        with mock.patch('intelmq.lib.utils.load_configuration',
                        new=self.mocked_config):
            with mock.patch('intelmq.lib.utils.log', self.get_mocked_logger(self.logger)):
                # insert the call to the update method here
                self.bot.stop(exitcode=0)
```

Reply:
Thanks! I updated the test with mocking.  All checks passed.